From 3e84e0999a2b61d9f0e04d3110a1561dd7f1b4b1 Mon Sep 17 00:00:00 2001
From: Andrew Boie <andrew.p.boie@intel.com>
Date: Wed, 25 Sep 2013 19:08:59 -0700
Subject: add userfastboot support

- Read the userfastboot Makefile which describes how to build
the images
- Add the boot image components to the target-files-package
- delete $OUT/userfastboot on 'make installclean'

Issue: ADEV-148
Issue: ADEV-21
Change-Id: I3f44bdac2d6a2bb3b1fda262fea5f22e92477fb1
Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>
---
 core/Makefile      |   28 ++++++++++++++++++++++++++++
 core/cleanbuild.mk |    1 +
 2 files changed, 29 insertions(+), 0 deletions(-)

diff --git a/core/Makefile b/core/Makefile
index 9e90b4b..ddc7c37 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -375,6 +375,10 @@ event-log-tags: $(event_log_tags_file)
 ALL_DEFAULT_INSTALLED_MODULES += $(event_log_tags_file)
 
 
+ifeq ($(strip $(TARGET_USE_USERFASTBOOT)),true)
+include bootable/userfastboot/ramdisk.mk
+endif
+
 # #################################################################
 # Targets for boot/OS images
 # #################################################################
@@ -1221,6 +1225,7 @@ $(BUILT_TARGET_FILES_PACKAGE): \
 		$(INSTALLED_VENDORIMAGE_TARGET) \
 		$(INSTALLED_ANDROID_INFO_TXT_TARGET) \
 		$(SELINUX_FC) \
+		$(USERFASTBOOT_BOOTIMAGE) \
 		$(built_ota_tools) \
 		$(APKCERTS_FILE) \
 		$(HOST_OUT_EXECUTABLES)/fs_config \
@@ -1268,6 +1273,26 @@ endif
 ifdef BOARD_KERNEL_PAGESIZE
 	$(hide) echo "$(BOARD_KERNEL_PAGESIZE)" > $(zip_root)/BOOT/pagesize
 endif
+ifeq ($(TARGET_STAGE_USERFASTBOOT),true)
+	@# Components of the fastboot image
+	$(hide) mkdir -p $(zip_root)/FASTBOOT
+	$(hide) $(call package_files-copy-root, \
+		$(USERFASTBOOT_ROOT_OUT),$(zip_root)/FASTBOOT/RAMDISK)
+ifdef INSTALLED_KERNEL_TARGET
+	$(hide) $(ACP) $(INSTALLED_KERNEL_TARGET) $(zip_root)/FASTBOOT/kernel
+endif
+ifdef INSTALLED_2NDBOOTLOADER_TARGET
+	$(hide) $(ACP) \
+		$(INSTALLED_2NDBOOTLOADER_TARGET) $(zip_root)/FASTBOOT/second
+endif
+	$(hide) echo "$(USERFASTBOOT_CMDLINE)" > $(zip_root)/FASTBOOT/cmdline
+ifdef BOARD_KERNEL_BASE
+	$(hide) echo "$(BOARD_KERNEL_BASE)" > $(zip_root)/FASTBOOT/base
+endif
+ifdef BOARD_KERNEL_PAGESIZE
+	$(hide) echo "$(BOARD_KERNEL_PAGESIZE)" > $(zip_root)/FASTBOOT/pagesize
+endif
+endif # TARGET_STAGE_USERFASTBOOT
 	$(hide) $(foreach t,$(INSTALLED_RADIOIMAGE_TARGET),\
 	            mkdir -p $(zip_root)/RADIO; \
 	            $(ACP) $(t) $(zip_root)/RADIO/$(notdir $(t));)
@@ -1317,6 +1342,9 @@ endif
 	$(hide) zipinfo -1 $@ | awk 'BEGIN { FS="SYSTEM/" } /^SYSTEM\// {print "system/" $$2}' | $(HOST_OUT_EXECUTABLES)/fs_config -C -S $(SELINUX_FC) > $(zip_root)/META/filesystem_config.txt
 	$(hide) zipinfo -1 $@ | awk 'BEGIN { FS="BOOT/RAMDISK/" } /^BOOT\/RAMDISK\// {print $$2}' | $(HOST_OUT_EXECUTABLES)/fs_config -C -S $(SELINUX_FC) > $(zip_root)/META/boot_filesystem_config.txt
 	$(hide) zipinfo -1 $@ | awk 'BEGIN { FS="RECOVERY/RAMDISK/" } /^RECOVERY\/RAMDISK\// {print $$2}' | $(HOST_OUT_EXECUTABLES)/fs_config -C -S $(SELINUX_FC) > $(zip_root)/META/recovery_filesystem_config.txt
+ifeq ($(TARGET_STAGE_USERFASTBOOT),true)
+	$(hide) zipinfo -1 $@ | awk 'BEGIN { FS="FASTBOOT/RAMDISK/" } /^FASTBOOT\/RAMDISK\// {print $$2}' | $(HOST_OUT_EXECUTABLES)/fs_config -C -S $(SELINUX_FC) > $(zip_root)/META/fastboot_filesystem_config.txt
+endif
 	$(hide) (cd $(zip_root) && zip -q ../$(notdir $@) META/*filesystem_config.txt)
 
 .PHONY: target-files-package
diff --git a/core/cleanbuild.mk b/core/cleanbuild.mk
index 2406616..7734958 100644
--- a/core/cleanbuild.mk
+++ b/core/cleanbuild.mk
@@ -192,6 +192,7 @@ installclean_files := \
 	$(PRODUCT_OUT)/recovery \
 	$(PRODUCT_OUT)/root \
 	$(PRODUCT_OUT)/system \
+	$(PRODUCT_OUT)/userfastboot \
 	$(PRODUCT_OUT)/dex_bootjars \
 	$(PRODUCT_OUT)/obj/JAVA_LIBRARIES \
 	$(PRODUCT_OUT)/obj/FAKE \
-- 
1.7.4.1

